src/ariel-os-stm32/Cargo.toml

```
@@ -30,6 +30,9 @@ ariel-os-embassy-common = { workspace = true }
 ariel-os-random = { workspace = true, optional = true }
 static_cell = { workspace = true }
 
+[build-dependencies]
+stm32-metapac = "*"
```

We have another dependency on `embassy-stm32` that choses a version. `*` here means `don't care`, so Cargo will chose the one `embassy-stm32` is fine with. Not sure if being more specific here makes any sense.

---

> `*` here means `don't care`, so Cargo will chose the one `embassy-stm32` is fine with

I'm not sure of that, it at least depends on the resolver; also [crates.io doesn't accept these](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#wildcard-requirements).

```suggestion
stm32-metapac = "16.0.0"
```